.github/workflows/linux.yml

```diff
@@ -108,3 +113,59 @@ jobs:
         if: matrix.cli == true
         run: sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v ${{ matrix.flags }} -s cli-suite
 
+  validate-old-ghcs:
+    name: Validate old ghcs ${{ matrix.extra-ghc }}
+    runs-on: ubuntu-18.04
+    needs: validate
+    container:
+      image: phadej/ghc:8.8.4-xenial
+
+    strategy:
+      matrix:
+        ghc: ["8.8.4"]
+        extra-ghc: ["7.10.3", "7.8.4", "7.6.3", "7.4.2", "7.2.2", "7.0.4"]
+
+    steps:
+
+      # We can't use actions/checkout with the docker container, see:
+      # https://github.com/actions/checkout/issues/170
+      # https://github.com/actions/checkout/issues/295
+      # - uses: actions/checkout@v2
+      - name: Checkout
+        run: |
+          echo $GITHUB_REF $GITHUB_SHA
+          git clone --depth 1 https://github.com/$GITHUB_REPOSITORY.git .
+          git fetch origin $GITHUB_SHA:temporary-ci-branch
+          git checkout $GITHUB_SHA || (git fetch && git checkout $GITHUB_SHA)
+
+      - name: Install extra compiler
+        run: |
+          apt-get update
+          apt-get install -y ghc-${{ matrix.extra-ghc }}-dyn
+
+      - uses: haskell/actions/setup@v1
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+
+      #  See the following link for a breakdown of the following step
+      #  https://github.com/haskell/actions/issues/7#issuecomment-745697160
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
+            dist-newstyle
+          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ github.sha }}
```

Comment:
We might need to change this to something else otherwise [this might happen](https://github.com/haskell/cabal/runs/5289177235?check_suite_focus=true#step:17:3)

```
Unable to reserve cache with key Linux-8.8.4-3553f69b8e2c836a6106ef937b70aba70a3dd7a5, another job may be creating this cache.
```

Reply:
hmm could it have any harmful consequence?
those jobs don't need update the cache, they are reusing the main ghc one from the previous step
ideally they should reuse the artifact generated by that previous step as noted above and don't even read it
I can add a comment about that refactoring

Reply:
can the job use the cache in only read mode? it would make clearer it's use until such refactoring

Reply:
You might be right in that it is not a problem. Let's break it down.

AFAIU all caches in GitHub are read-only. Once they are made and associated with a key, they are not changed.

The commit was 3553f69b8e2c836a6106ef937b70aba70a3dd7a5.
1) They [fetch](https://github.com/haskell/cabal/runs/5289177235?check_suite_focus=true#step:6:14) the most recent compatible key (`Linux-8.8.4-a45a404a339570459248a81f96d41feb5f6e6922` given we give `Linux-8.8.4-` as a prefix)
2) They do their work, possibly compiling new depdendencies
3) They try to save the new cache with key `Linux-8.8.4-3553f69b8e2c836a6106ef937b70aba70a3dd7a5` and here they fail because two jobs are saving the same key at the same time.

Now, if we assume the two jobs have updated the cache in the same way, perhaps it's fine, only one needs to be able to save it.

Reply:
> can the job use the cache in only read mode? it would make clearer its use until such refactoring

it seems it is not possible: https://github.com/actions/cache/issues/350